### ISSUE TITLE
Fastsim era change causes CkfBaseTrajectoryFilter_block to be a problem

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#21267]
The workflows 135.8 and 250399.0 are failing in cmsDriver.py with error
```
Traceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/bin/slc6_amd64_gcc630/cmsDriver.py", line 55, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/bin/slc6_amd64_gcc630/cmsDriver.py", line 27, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/python/Configuration/Applications/ConfigBuilder.py", line 2271, in prepare
    self.process = customiseEarlyDelete(self.process)
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/python/Configuration/StandardSequences/earlyDeleteSettings_cff.py", line 77, in customiseEarlyDelete
    _hasInputTagModuleLabel(process, module, producers,result)
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/python/Configuration/StandardSequences/earlyDeleteSettings_cff.py", line 14, in _hasInputTagModuleLabel
    _hasInputTagModuleLabel(process, value, moduleLabels,result)
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/python/Configuration/StandardSequences/earlyDeleteSettings_cff.py", line 33, in _hasInputTagModuleLabel
    _hasInputTagModuleLabel(process, getattr(process, value.value()), moduleLabels,result)
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/python/Configuration/StandardSequences/earlyDeleteSettings_cff.py", line 14, in _hasInputTagModuleLabel
    _hasInputTagModuleLabel(process, value, moduleLabels,result)
  File "/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_0_X_2017-11-09-2300/python/Configuration/StandardSequences/earlyDeleteSettings_cff.py", line 33, in _hasInputTagModuleLabel
    _hasInputTagModuleLabel(process, getattr(process, value.value()), moduleLabels,result)
AttributeError: 'Process' object has no attribute 'CkfBaseTrajectoryFilter_block'
```

**Comment by Dr15Jones** [comment]
assign fastsim

**Comment by Dr15Jones** [comment]
One of the changes when doing the fastsim era migration was to not remove modules from the reconstruction python modules when loading fastsim. So these modules are added to the `cms.Process` but are associated to any running `cms.Path` or `cms.EndPath` and therefore will not be used in cmsRun. However, there configuration fragments will still be parsed which means any outstanding `refToPSets` will get resolved. It is during this resolution step that things are failing.

**Comment by Dr15Jones** [comment]
I'm looking further into this to see how it can be fixed.

**Comment by Dr15Jones** [comment]
It appears that workflows 135.8 and 250399.0 differ from the other fastsim workflows in that they do step `RECOBEFMIX` but do not additional do the step `RECO`. This is relevant to this problem since the module which is failing is `jetCoreRegionalStepTrajectoryBuilder` which gets loaded from 
`FastSimulation.Configuration.Reconstruction_BefMix_cff`
and the module has a `refToPSet_` to `CkfBaseTrajectoryFilter_block` which is loaded into the process at the load of `FastSimulation.Configuration.Reconstruction_AftMix_cff`. This latter load does not happen if one only does `RECOBEFMIX`.

**Comment by Dr15Jones** [comment]
I appear to have a fix for this. I am testing it further.

**Comment by Dr15Jones** [comment]
This is fixed with #21268

**Comment by Dr15Jones** [comment]
assign reconstruction

**Comment by Dr15Jones** [comment]
I added reconstruction to this issue since the fix was to change configurations that are in their domain.

**Comment by slava77** [comment]
+1

the crash reported in the issue description is fixed based on #21268, which has been merged in 100X
